### PR TITLE
chore: drop Node 12 support

### PR DESCRIPTION
**Pull request by cat2608**
#### What does this PR do?

This PR updates Nodejs version from 12 to 14.

- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### Any background context you want to provide?
- https://github.com/snyk/cli/pull/1042
- https://github.com/snyk/cli/pull/1639

#### What are the relevant tickets?

HEAD-496

### CODE REVIEW

File: release-scripts/snyk-mac.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 DIRNAME=$(dirname "$0")
 
-NODE="$DIRNAME/node-v12.22.11-darwin-x64/bin/node"
+NODE="$DIRNAME/node-v14.21.3-darwin-x64/bin/node"
```
Comment (PeterSchafer) [inline release-scripts/snyk-mac.sh:6]: Suggestion: as we stopped deploying these builds and are about to cleanup the files, let's undo these changes.

File: release-scripts/docker-desktop-release.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ cp ./release-scripts/snyk-mac.sh ./dist-docker/docker/
 cd ./dist-docker/docker/
 
 # Download macOS NodeJS binary, using same as pkg
-curl "https://nodejs.org/dist/v12.22.11/node-v12.22.11-darwin-x64.tar.gz" | tar -xz
+curl "https://nodejs.org/dist/v14.21.3/node-v14.21.3-darwin-x64.tar.gz" | tar -xz
```
Comment (PeterSchafer) [inline release-scripts/docker-desktop-release.sh:20]: Suggestion: as we stopped deploying these builds and are about to cleanup the files, let's undo these changes.

**Review by bastiandoetsch — APPROVED** [review]
LGTM 🎉
